examples/lab_devices/keithley6517a.py

```diff
@@ -0,0 +1,76 @@
+###################################################
+# This is an example for keithley6517a electrometer
+# Last Modified: Mi 01 Jun 2022 07:30:23  CEST
+###################################################
+
+import time
+from basil.dut import Dut
+from basil.TL.Visa import visa
+
+##################
+# Initialisation #
+##################
+
+dut = Dut('keithley6517a_pyvisa.yaml')
+dut.init()
+time.sleep(0.3)
+
+#########################
+# End of Initialisation #
+#########################
+
+#############################
+# Settings for Electrometer #
+#############################
+
+# check and configure the filter
+time.sleep(0.5)
+dut['EMeter'].set_Iavefilter_type('REP')            # "REP" means repeating
+dut['EMeter'].set_Iavefilter_count('10')            # take 10 values for averaging
+time.sleep(0.5)
+dut['EMeter'].I_filter_on()                           # turn on the filter
+print("Filter : ", dut['EMeter'].get_I_filter())      # display the status of the filter
+time.sleep(0.5)
+avecou = int(dut['EMeter'].get_Iavefilter_count())  # display the number of values for averaging
+print('averaging over:', avecou)
+time.sleep(0.5)
+
+# set trigger status
+dut['EMeter'].trigger_conti_off()                   # turn off the continuous trigger
+time.sleep(0.5)
+
+time.sleep(0.3)
+dut['EMeter'].select_current()          # measure current
+dut['EMeter'].connect_meter()           # connect the source and meter, resulting in a simpler circuit
+dut['EMeter'].set_source_range('MAX')   # set the output limit of the voltage source to 1000V (MAX)
+
+# Perform zero check
+# set the smallest range for zero check
+dut['EMeter'].set_current_range(20e-12)     # unit is A (everywhere)
+time.sleep(0.5)
```

Comment:
Does the manual say anything about waiting times for certain operations?

Reply:
Can you check something like an `is_ready` parameter on the device instead?

Reply:
I didn't notice about a standard setting of the waiting time. So these are empirical, set the time until it stops complaining. maybe at some points I set a slightly too long time, but will work

Reply:
> Can you check something like an `is_ready` parameter on the device instead?

Is there any device using this feature, it would be faster to check what command it uses and then look for the similar commands, as long as it's SCPI

Reply:
There is something in SCPI maybe you can check the two options mentioned [here](https://stackoverflow.com/questions/30725746/is-it-normal-to-wait-for-some-time-between-operations-under-the-visa-protocol)

Reply:
@s6thseng maybe the solution in the link can help you to sync your R&S PSU so you don't have to put arbitrary sleeps and/or try many times

Reply:
The suggested SCPI commands are "*WAI" and "*OPC".
I've tried both commands and they doesn't seems working. According to the manual, these 2 commands are especially useful for the so called "overlapped commands":
![Screenshot 2022-06-02 at 13 48 57](https://user-images.githubusercontent.com/31743227/171629680-db4e0489-9e26-4457-84a5-c670d2bf810f.png)
The test I've done purely consists of the other type (sequential commands), which should not need the "waiting" commands according to the manual. As a result of the test, error messages showed up as if no "waiting" commands were applied.
I've also tested the possibility of wrong command format, by adding an additional time sleep. No error message has appeared.
I guess there are probably a smarter way to do this "waiting" procedure, but it seems less obvious.
For now, I would say that the time sleep could be a good alternative, since the setting only needs to be made once in the script.

BY THE WAY, please don't merge them now, I found some mistakes in the example, I will fix them first

Reply:
Tried your proposed solution, but still does not work. To sleep is still the easiest solution.

Reply:
For VISA you can set a `query_delay` in the transfer layer config's `init`, can you test if this removes the need for manual sleeps?

Reply:
seems not to work for the PS